baseapp/baseapp.go

```diff
@@ -678,15 +686,35 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 	if err == nil && mode == runTxModeDeliver {
-		msCache.Write()
-
-		if len(anteEvents) > 0 {
-			// append the events in the order of occurrence
-			result.Events = append(anteEvents, result.Events...)
```

Comment:
we definitely need the anteEvents added here?

Reply:
can someone re-review the ante handler  events being propagated correctly.

Reply:
@channa-figure Could you?